internal/indexmanagement/scripts.go

```diff
@@ -131,6 +131,92 @@ def updateWriteIndex(currentIndex, nextIndex, alias):
     print(e)
     sys.stdout = original_stdout
     return False
+
+def getMaxAllowedSize(diskThreshold):
+  # Returns the total disk space of all nodes combined multiplied by the pre-defined threshold
+  original_stdout = sys.stdout
+  try:
+    es_client = getEsClient()
+    response = es_client.cat.allocation(format="JSON", h="disk.total", bytes="kb")
+    totalSize = 0
+    for i in range(3):
+      #print (response[i]["disk.total"])
+      totalSize = totalSize + int(response[i]["disk.total"])
+    return int(diskThreshold/100.0 * totalSize)
+  except Exception as e:
+    sys.stdout = open('/tmp/response.txt', 'w')
+    print(e)
+    sys.stdout = original_stdout
+    return -1
+
+def getDeletableIndices(index, alias, maxAllowedSize, indicesSizeCounter, indicesToDelete):
+  # Returns a list of indices from [alias] (infra, app, or audit), that should be deleted
+  # traverse through indices from newest to oldest, adding up their size. When the sum of sizes exceeds
+  # maxAllowedSize, start adding indices to the list
+  original_stdout = sys.stdout
+  try:
+    expectedSize = int(index['store.size']) + indicesSizeCounter
+    print ("expected size = ", expectedSize, "KB")
+    if indicesSizeCounter < maxAllowedSize and expectedSize < maxAllowedSize:
+      print ("oooooooooo", index['index'], "stays.")
+      indicesSizeCounter += int(index['store.size'])
+    else:
+      print (index['index'], "'s size is starting to get over threshold")
+      if index['index'].startswith(alias):
+        print ("----------", index['index'], "should be deleted")
+        indicesToDelete.append(index['index'])
+    return indicesToDelete, indicesSizeCounter
+  except Exception as e:
+    #sys.stdout = open('/tmp/response.txt', 'w')
+    print(e)
+    sys.stdout = original_stdout
+    return -1
```

Comment:
What if we flip the traversal here from oldest to newest? This should simplify things a lot, because by default we append indices (starting from oldest) until the sum of the appended indices hits the threshold. WDYT?

Reply:
After doing a few more tests, I figured that the current way is the only one to prevent over-deletion of indices.
Earlier, I saw some indices being deleted each time the cron job was running. That actually wasn't over-deletion, since we are counting up the index sizes from newest to oldest, until the size reaches the threshold. So each time an old index was deleted, it was because the threshold was actually exceeded since new indices were also being created (due to rollover), thus taking up more space.
For example, if we have 100GB of total disk space for all nodes, and the threshold is 70%, we add up index sizes starting from the most recent index. When the sum exceeds 70GB, we start deleting indices, which would be the oldest ones of course. 
In conclusion, the way that lightweight-curator was implemented in the first place was to prevent over-deletion.